In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sb
pd.set_option('display.max_rows', 50)
alcoholEstudiantesMat = pd.read_csv("/Users/eduardogonzalez/Documents/examen2/student-mat.csv")
alcoholEstudiantesPor= pd.read_csv("/Users/eduardogonzalez/Documents/examen2/student-por.csv")
from collections import Counter
from datetime import date, timedelta, datetime
import plotly.graph_objs as go
from scipy.stats import ttest_1samp, wilcoxon, ttest_ind, mannwhitneyu
import numpy as np 

In [2]:
alcoholEstudiantesMat.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 395 entries, 0 to 394
Data columns (total 33 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   school      395 non-null    object
 1   sex         395 non-null    object
 2   age         395 non-null    int64 
 3   address     395 non-null    object
 4   famsize     395 non-null    object
 5   Pstatus     395 non-null    object
 6   Medu        395 non-null    int64 
 7   Fedu        395 non-null    int64 
 8   Mjob        395 non-null    object
 9   Fjob        395 non-null    object
 10  reason      395 non-null    object
 11  guardian    395 non-null    object
 12  traveltime  395 non-null    int64 
 13  studytime   395 non-null    int64 
 14  failures    395 non-null    int64 
 15  schoolsup   395 non-null    object
 16  famsup      395 non-null    object
 17  paid        395 non-null    object
 18  activities  395 non-null    object
 19  nursery     395 non-null    object
 20  higher    

In [3]:
alcoholEstudiantesMat.head()

school sex  age address famsize Pstatus  Medu  Fedu     Mjob      Fjob  ...  \
0     GP   F   18       U     GT3       A     4     4  at_home   teacher  ...   
1     GP   F   17       U     GT3       T     1     1  at_home     other  ...   
2     GP   F   15       U     LE3       T     1     1  at_home     other  ...   
3     GP   F   15       U     GT3       T     4     2   health  services  ...   
4     GP   F   16       U     GT3       T     3     3    other     other  ...   

  famrel freetime  goout  Dalc  Walc health absences  G1  G2  G3  
0      4        3      4     1     1      3        6   5   6   6  
1      5        3      3     1     1      3        4   5   5   6  
2      4        3      2     2     3      3       10   7   8  10  
3      3        2      2     1     1      5        2  15  14  15  
4      4        3      2     1     2      5        4   6  10  10  

[5 rows x 33 columns]

In [8]:
alcoholEstudiantesMat.describe()

age        Medu        Fedu  traveltime   studytime    failures  \
count  395.000000  395.000000  395.000000  395.000000  395.000000  395.000000   
mean    16.696203    2.749367    2.521519    1.448101    2.035443    0.334177   
std      1.276043    1.094735    1.088201    0.697505    0.839240    0.743651   
min     15.000000    0.000000    0.000000    1.000000    1.000000    0.000000   
25%     16.000000    2.000000    2.000000    1.000000    1.000000    0.000000   
50%     17.000000    3.000000    2.000000    1.000000    2.000000    0.000000   
75%     18.000000    4.000000    3.000000    2.000000    2.000000    0.000000   
max     22.000000    4.000000    4.000000    4.000000    4.000000    3.000000   

           famrel    freetime       goout        Dalc        Walc      health  \
count  395.000000  395.000000  395.000000  395.000000  395.000000  395.000000   
mean     3.944304    3.235443    3.108861    1.481013    2.291139    3.554430   
std      0.896659    0.998862    1.113278    0.890741    1.287897    1.390303   
min      1.000000    1.000000    1.000000    1.000000    1.000000    1.000000   
25%      4.000000    3.000000    2.000000    1.000000    1.000000    3.000000   
50%      4.000000    3.000000    3.000000    1.000000    2.000000    4.000000   
75%      5.000000    4.000000    4.000000    2.000000    3.000000    5.000000   
max      5.000000    5.000000    5.000000    5.000000    5.000000    5.000000   

         absences          G1          G2          G3  
count  395.000000  395.000000  395.000000  395.000000  
mean     5.708861   10.908861   10.713924   10.415190  
std      8.003096    3.319195    3.761505    4.581443  
min      0.000000    3.000000    0.000000    0.000000  
25%      0.000000    8.000000    9.000000    8.000000  
50%      4.000000   11.000000   11.000000   11.000000  
75%      8.000000   13.000000   13.000000   14.000000  
max     75.000000   19.000000   19.000000   20.000000

In [9]:
alcoholEstudiantesPor.describe()

age        Medu        Fedu  traveltime   studytime    failures  \
count  649.000000  649.000000  649.000000  649.000000  649.000000  649.000000   
mean    16.744222    2.514638    2.306626    1.568567    1.930663    0.221880   
std      1.218138    1.134552    1.099931    0.748660    0.829510    0.593235   
min     15.000000    0.000000    0.000000    1.000000    1.000000    0.000000   
25%     16.000000    2.000000    1.000000    1.000000    1.000000    0.000000   
50%     17.000000    2.000000    2.000000    1.000000    2.000000    0.000000   
75%     18.000000    4.000000    3.000000    2.000000    2.000000    0.000000   
max     22.000000    4.000000    4.000000    4.000000    4.000000    3.000000   

           famrel    freetime       goout        Dalc        Walc      health  \
count  649.000000  649.000000  649.000000  649.000000  649.000000  649.000000   
mean     3.930663    3.180277    3.184900    1.502311    2.280431    3.536210   
std      0.955717    1.051093    1.175766    0.924834    1.284380    1.446259   
min      1.000000    1.000000    1.000000    1.000000    1.000000    1.000000   
25%      4.000000    3.000000    2.000000    1.000000    1.000000    2.000000   
50%      4.000000    3.000000    3.000000    1.000000    2.000000    4.000000   
75%      5.000000    4.000000    4.000000    2.000000    3.000000    5.000000   
max      5.000000    5.000000    5.000000    5.000000    5.000000    5.000000   

         absences          G1          G2          G3  
count  649.000000  649.000000  649.000000  649.000000  
mean     3.659476   11.399076   11.570108   11.906009  
std      4.640759    2.745265    2.913639    3.230656  
min      0.000000    0.000000    0.000000    0.000000  
25%      0.000000   10.000000   10.000000   10.000000  
50%      2.000000   11.000000   11.000000   12.000000  
75%      6.000000   13.000000   13.000000   14.000000  
max     32.000000   19.000000   19.000000   19.000000

# cantidad de alcohol ingerido en fines de semana clase matematicas


In [10]:
import plotly.express as px
fig = px.histogram(alcoholEstudiantesMat, x="age", color="Dalc", marginal="violin", hover_data=alcoholEstudiantesMat.columns, 
                   title ="Distribution of AGE Vs Alcohol en Matematicas", 
                   labels={"age": "AGE"},
                   color_discrete_map={"0": "RebeccaPurple", "1": "MediumPurple",},
                   width=720, height=480
                  ) 
fig.show()

# cantidad de alcohol ingerido en fines de semana clase Portugües

In [22]:
import plotly.express as px
fig = px.histogram(alcoholEstudiantesPor, x="age", color="Dalc", marginal="violin", hover_data=alcoholEstudiantesPor.columns, 
                   title ="Distribution of AGE Vs Alcohol en Portugues", 
                   labels={"age": "AGE"},
                   color_discrete_map={"0": "RebeccaPurple", "1": "MediumPurple",},
                   width=720, height=480
                  ) 
fig.show()

# Primer t-test

In [14]:
from scipy.stats import ttest_ind
ConsumoEntreSemanaMat = alcoholEstudiantesMat['Dalc']
ConsumoEntreSemanaPor = alcoholEstudiantesPor['Dalc']

stat, p = ttest_ind(ConsumoEntreSemanaMat, ConsumoEntreSemanaPor)
print('stat=%.3f, p=%.3f' % (stat, p))
if p > 0.05:
	print('Probably the same distribution')
else:
	print('Probably different distributions')

stat=-0.366, p=0.715
Probably the same distribution


## Vamos a tomar el famoso umbral de significancia como p = 0.05 Al realizar nuestras pruebas nos vamos a enfocar en el valor P que nos da nuestros calculos, es decir al ingresar los datos a la función correspondiente nos enfocaremos en P 

### Como se puede ver al correr el codigo con los datos correspondientes, vemos que p nos da un valor de 0.715 esto quiere decir que nuestra p calculada es mucho mayor a nuestro umbral de significancia que habiamos establecido previamente esto quiere decir que nuestra hipotesis nula no se rechaza, es decir, podemos decir que no existe una diferencia entre la clase de matematicas y portugués viendolo desde el punto de vista de el consumo de alcohol en días laborales de los alumnos

## Cantidad de clases reprobadas vs Edad en clase Matemáticas

In [15]:
import plotly.express as px
fig = px.histogram(alcoholEstudiantesMat, x="Walc", color="failures", marginal="violin", hover_data=alcoholEstudiantesMat.columns, 
                   title ="Distribution of Walc Vs failures en Matematicas", 
                   labels={"A": "walc"},
                   color_discrete_map={"0": "RebeccaPurple", "1": "MediumPurple",},
                   width=720, height=480
                  ) 
fig.show()

# Segundo t-test

In [16]:
CalificacionFinalMat = alcoholEstudiantesMat['failures']
CalificacionFinalPor = alcoholEstudiantesPor['failures']

stat, p = ttest_ind(CalificacionFinalMat, CalificacionFinalPor)
print('stat=%.3f, p=%.9f' % (stat, p))

stat=2.690, p=0.007261460


## Cantidad de clases reprobadas vs Edad en clase Portugués

In [21]:
import plotly.express as px
fig = px.histogram(alcoholEstudiantesPor, x="age", color="failures", marginal="violin", hover_data=alcoholEstudiantesPor.columns, 
                   title ="Distribution of AGE Vs Alcohol en Portugués", 
                   labels={"age": "AGE"},
                   color_discrete_map={"0": "RebeccaPurple", "1": "MediumPurple",},
                   width=720, height=480
                  ) 
fig.show()

## Cantidad de clases reprobadas vs Edad en clase Matemáticas

In [18]:
import plotly.express as px
fig = px.histogram(alcoholEstudiantesMat, x="age", color="failures", marginal="violin", hover_data=alcoholEstudiantesMat.columns, 
                   title ="Distribution of AGE Vs Alcohol en Matématicas", 
                   labels={"age": "AGE"},
                   color_discrete_map={"0": "RebeccaPurple", "1": "MediumPurple",},
                   width=720, height=480
                  ) 
fig.show()

### Ahora haremos una comparación de las clases , de las veces que reprobaron dichas clases anteriormente,vamos hacer el mismo análisis con mismas pruebas, pero enfoque distinto, nuestra hipotesis nula esta vez se enfocara en ver que no hay diferencia entre las veces que han reprobado anteriormente las respectivas clases, y la hipotesis alterna es que si hay diferencia entre las dos clases 

### Como se puede ver al correr el codigo con los datos correspondientes, vemos que p nos da un valor de 0.007261 lo cual es mucho menor que el umbral de significancia, lo que nos lleva a rechazar nuestra hipotesis nula y concluimos que si existe diferencia entre las clases de matématicas y portugués tomando en cuanta las veces que sus alumnos han reprobado anteriormente la dicha materia
